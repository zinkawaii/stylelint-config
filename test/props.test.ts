import { all } from "known-css-properties";
import { expect, it } from "vitest";
import config from "../src/lib/order";

it("check props", () => {
    const logical = /(?:block|inline)-(?:start|end)/;

    const filtered = all.filter((name) => {
        return !name.startsWith("-") && !logical.test(name);
    });

    const flattened = config["order/properties-order"].flatMap((group) => {
        return typeof group === "object" ? group.properties : group;
    });

    const missings = filtered.filter((prop) => !flattened.includes(prop));
    const mistakens = flattened.filter((prop) => !filtered.includes(prop));

    expect(missings).matchSnapshot("missings");
    expect(mistakens).matchSnapshot("mistakens");
});
