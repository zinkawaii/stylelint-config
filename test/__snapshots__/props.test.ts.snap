// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`check props > missings 1`] = `
[
  "accelerator",
  "additive-symbols",
  "alignment-baseline",
  "alt",
  "anchor-name",
  "anchor-scope",
  "animation-trigger",
  "animation-trigger-behavior",
  "animation-trigger-exit-range",
  "animation-trigger-exit-range-end",
  "animation-trigger-exit-range-start",
  "animation-trigger-range",
  "animation-trigger-range-end",
  "animation-trigger-range-start",
  "animation-trigger-timeline",
  "app-region",
  "ascent-override",
  "audio-level",
  "azimuth",
  "background-repeat-x",
  "background-repeat-y",
  "base-palette",
  "baseline-shift",
  "baseline-source",
  "behavior",
  "block-ellipsis",
  "block-step",
  "block-step-align",
  "block-step-insert",
  "block-step-round",
  "block-step-size",
  "bookmark-label",
  "bookmark-level",
  "bookmark-state",
  "border-boundary",
  "border-end-end-radius",
  "border-end-start-radius",
  "border-start-end-radius",
  "border-start-start-radius",
  "box-snap",
  "break-after",
  "break-before",
  "break-inside",
  "buffered-rendering",
  "caret",
  "caret-animation",
  "caret-shape",
  "chains",
  "clip",
  "clip-rule",
  "color-adjust",
  "color-interpolation-filters",
  "color-profile",
  "color-rendering",
  "column-height",
  "column-progression",
  "column-rule-break",
  "column-rule-outset",
  "column-wrap",
  "contain-intrinsic-block-size",
  "contain-intrinsic-height",
  "contain-intrinsic-inline-size",
  "contain-intrinsic-size",
  "contain-intrinsic-width",
  "content-visibility",
  "continue",
  "cue",
  "cue-after",
  "cue-before",
  "cx",
  "cy",
  "d",
  "descent-override",
  "display-align",
  "dominant-baseline",
  "dynamic-range-limit",
  "elevation",
  "enable-background",
  "epub-caption-side",
  "epub-hyphens",
  "epub-text-combine",
  "epub-text-emphasis",
  "epub-text-emphasis-color",
  "epub-text-emphasis-style",
  "epub-text-orientation",
  "epub-text-transform",
  "epub-word-break",
  "epub-writing-mode",
  "fallback",
  "fill",
  "fill-break",
  "fill-color",
  "fill-image",
  "fill-opacity",
  "fill-origin",
  "fill-position",
  "fill-repeat",
  "fill-rule",
  "fill-size",
  "float-defer",
  "float-offset",
  "float-reference",
  "flood-color",
  "flood-opacity",
  "flow",
  "flow-from",
  "flow-into",
  "font-display",
  "font-synthesis-position",
  "font-synthesis-small-caps",
  "font-synthesis-style",
  "font-synthesis-weight",
  "font-variant-alternates",
  "font-variant-caps",
  "font-variant-east-asian",
  "font-variant-emoji",
  "font-variant-ligatures",
  "font-variant-numeric",
  "font-variant-position",
  "font-variation-settings",
  "font-width",
  "footnote-display",
  "footnote-policy",
  "glyph-orientation-horizontal",
  "glyph-orientation-vertical",
  "hyphenate-limit-last",
  "hyphenate-limit-lines",
  "hyphenate-limit-zone",
  "ime-mode",
  "inherits",
  "initial-letter-align",
  "initial-letter-wrap",
  "initial-value",
  "inline-sizing",
  "input-format",
  "input-security",
  "inset-area",
  "inset-block",
  "inset-inline",
  "interactivity",
  "interpolate-size",
  "item-cross",
  "item-direction",
  "item-flow",
  "item-pack",
  "item-slack",
  "item-track",
  "item-wrap",
  "kerning",
  "layout-flow",
  "layout-grid",
  "layout-grid-char",
  "layout-grid-line",
  "layout-grid-mode",
  "layout-grid-type",
  "lighting-color",
  "line-fit-edge",
  "line-gap-override",
  "line-grid",
  "line-height-step",
  "line-increment",
  "line-padding",
  "line-snap",
  "margin-break",
  "marker",
  "marker-end",
  "marker-knockout-left",
  "marker-knockout-right",
  "marker-mid",
  "marker-offset",
  "marker-pattern",
  "marker-segment",
  "marker-side",
  "marker-start",
  "marks",
  "mask-position-x",
  "mask-position-y",
  "mask-source-type",
  "math-depth",
  "math-shift",
  "math-style",
  "max-lines",
  "max-zoom",
  "min-intrinsic-sizing",
  "min-zoom",
  "motion",
  "motion-offset",
  "motion-path",
  "motion-rotation",
  "nav-down",
  "nav-index",
  "nav-left",
  "nav-right",
  "nav-up",
  "navigation",
  "negative",
  "object-view-box",
  "offset-rotation",
  "orientation",
  "overflow-clip-margin-block",
  "overflow-clip-margin-bottom",
  "overflow-clip-margin-inline",
  "overflow-clip-margin-left",
  "overflow-clip-margin-right",
  "overflow-clip-margin-top",
  "overlay",
  "override-colors",
  "pad",
  "page",
  "page-break-after",
  "page-break-before",
  "page-break-inside",
  "page-orientation",
  "pause",
  "pause-after",
  "pause-before",
  "pen-action",
  "perspective-origin-x",
  "perspective-origin-y",
  "pitch",
  "pitch-range",
  "play-during",
  "position-anchor",
  "position-area",
  "position-try",
  "position-try-fallbacks",
  "position-try-options",
  "position-try-order",
  "position-visibility",
  "prefix",
  "property-name",
  "r",
  "range",
  "reading-flow",
  "reading-order",
  "region-fragment",
  "rest",
  "rest-after",
  "rest-before",
  "richness",
  "row-rule",
  "row-rule-break",
  "row-rule-color",
  "row-rule-outset",
  "row-rule-style",
  "row-rule-width",
  "ruby-merge",
  "ruby-overhang",
  "rule",
  "rule-break",
  "rule-color",
  "rule-outset",
  "rule-paint-order",
  "rule-style",
  "rule-width",
  "running",
  "rx",
  "ry",
  "scroll-initial-target",
  "scroll-snap-coordinate",
  "scroll-snap-destination",
  "scroll-snap-margin",
  "scroll-snap-margin-bottom",
  "scroll-snap-margin-left",
  "scroll-snap-margin-right",
  "scroll-snap-margin-top",
  "scroll-snap-points-x",
  "scroll-snap-points-y",
  "scroll-snap-type-x",
  "scroll-snap-type-y",
  "scroll-start-target",
  "scroll-target-group",
  "scrollbar-arrow-color",
  "scrollbar-base-color",
  "scrollbar-dark-shadow-color",
  "scrollbar-darkshadow-color",
  "scrollbar-face-color",
  "scrollbar-highlight-color",
  "scrollbar-shadow-color",
  "scrollbar-track-color",
  "scrollbar3d-light-color",
  "scrollbar3dlight-color",
  "shape-inside",
  "shape-rendering",
  "size",
  "size-adjust",
  "slider-orientation",
  "snap-height",
  "solid-color",
  "solid-opacity",
  "spatial-navigation-action",
  "spatial-navigation-contain",
  "spatial-navigation-function",
  "speak",
  "speak-as",
  "speak-header",
  "speak-numeral",
  "speak-punctuation",
  "speech-rate",
  "src",
  "stop-color",
  "stop-opacity",
  "stress",
  "string-set",
  "stroke",
  "stroke-align",
  "stroke-alignment",
  "stroke-break",
  "stroke-color",
  "stroke-dash-corner",
  "stroke-dash-justify",
  "stroke-dashadjust",
  "stroke-dasharray",
  "stroke-dashcorner",
  "stroke-dashoffset",
  "stroke-image",
  "stroke-linecap",
  "stroke-linejoin",
  "stroke-miterlimit",
  "stroke-opacity",
  "stroke-origin",
  "stroke-position",
  "stroke-repeat",
  "stroke-size",
  "stroke-width",
  "suffix",
  "supported-color-schemes",
  "symbols",
  "syntax",
  "system",
  "text-align-all",
  "text-anchor",
  "text-decoration-blink",
  "text-decoration-line-through",
  "text-decoration-none",
  "text-decoration-overline",
  "text-decoration-skip",
  "text-decoration-skip-box",
  "text-decoration-skip-inset",
  "text-decoration-skip-self",
  "text-decoration-skip-spaces",
  "text-decoration-trim",
  "text-decoration-underline",
  "text-emphasis-skip",
  "text-group-align",
  "text-justify-trim",
  "text-kashida",
  "text-kashida-space",
  "text-line-through",
  "text-line-through-color",
  "text-line-through-mode",
  "text-line-through-style",
  "text-line-through-width",
  "text-overline",
  "text-overline-color",
  "text-overline-mode",
  "text-overline-style",
  "text-overline-width",
  "text-spacing",
  "text-underline",
  "text-underline-color",
  "text-underline-mode",
  "text-underline-offset",
  "text-underline-position",
  "text-underline-style",
  "text-underline-width",
  "text-wrap-mode",
  "text-wrap-style",
  "timeline-scope",
  "touch-action-delay",
  "transform-origin-x",
  "transform-origin-y",
  "transform-origin-z",
  "types",
  "uc-alt-skin",
  "uc-skin",
  "unicode-range",
  "user-zoom",
  "vector-effect",
  "view-transition-class",
  "view-transition-group",
  "viewport-fill",
  "viewport-fill-opacity",
  "viewport-fit",
  "voice-balance",
  "voice-duration",
  "voice-family",
  "voice-pitch",
  "voice-range",
  "voice-rate",
  "voice-stress",
  "voice-volume",
  "volume",
  "white-space-trim",
  "widows",
  "word-space-transform",
  "wrap-after",
  "wrap-before",
  "wrap-flow",
  "wrap-inside",
  "wrap-through",
  "x",
  "y",
]
`;

exports[`check props > mistakens 1`] = `[]`;
